Sage/carbon
# fix(filterable-select, simple-select): ensure an ellipsis is shown during text overflow

fix #6641

### Proposed behaviour



<img width="886" alt="Screenshot 2024-04-15 at 16 04 32" src="https://github.com/Sage/carbon/assets/114918852/af50701b-dbb7-465e-9a27-51d26580179c">

Ensures that an ellipsis is applied when text overflows, to make users fully aware that truncation is taking place

### Current behaviour



<img width="874" alt="Screenshot 2024-04-15 at 16 06 56" src="https://github.com/Sage/carbon/assets/114918852/0aa54e19-4d27-41e9-ac2c-82021cc99d3e">

Currently when text truncates no ellipsis is applied, meaning some users are not fully aware truncation is actually taking place

### Checklist




#### QA


### Additional context



### Testing instructions



Check the new test stories added for `FilterableSelect` and `SimpleSelect` to ensure that truncation occurs on both components
